### PR TITLE
[feat] Added distribution mapper

### PR DESCRIPTION
**Pull request by shijianjian**
It will aid the customisation of distributions in augmentation random generators.

### CODE REVIEW

File: kornia/augmentation/random_generator/base.py [hunk 1]
```diff
@@ -37,3 +38,57 @@ def make_samplers(self, device: torch.device, dtype: torch.dtype) -> None:
...
+    Example:
+        >>> from torch.distributions import Normal
+        >>> import torch.nn as nn
+        >>> # without mapper
+        >>> dist = DistributionWithMapper(Normal(0., 1.,), mapper=None)
+        >>> _ = torch.manual_seed(0)
+        >>> dist.rsample((8,))
+        tensor([ 1.5410, -0.2934, -2.1788,  0.5684, -1.0845, -1.3986,  0.4033,  0.8380])
+        >>> # with sigmoid mapper
+        >>> dist = DistributionWithMapper(Normal(0., 1.,), mapper=nn.Sigmoid())
```
Comment (edgarriba) [inline kornia/augmentation/random_generator/base.py:62]: just to get it right -- this applies Normal and then sigmoid ?

Comment (shijianjian) [reply kornia/augmentation/random_generator/base.py:62]: Yes. I will rename it to value_mapper to make it more clear.

File: kornia/augmentation/random_generator/base.py [hunk 2]
```diff
@@ -48,43 +48,43 @@ class DistributionWithMapper(Distribution):
 
     Args:
         dist: the target distribution.
-        mapper: the function or module to adjust the output from distributions.
+        value_mapper: the function or module to adjust the output from distributions.
```
Comment (edgarriba) [inline kornia/augmentation/random_generator/base.py:51]: `map_fcn` ?

**Review by edgarriba — APPROVED** [review]
